packages/components/organisms/f-form/src/components/FormField.vue

```diff
@@ -0,0 +1,150 @@
+        @blur="hasInvalidErrorMessage && formFieldBlur()">
+        <template
+            v-if="hasError"
+            #error>
+            <error-message
+                :id="translations.errorName"
+                data-js-error-message
+                :data-test-id="translations[`${errorType}Error`]">
+                {{ translations.validationMessages[errorType] }}
+            </error-message>
+        </template>
+    </form-field>
+</template>
+
+<script>
+import FormField from '@justeat/f-form-field';
+import ErrorMessage from '@justeat/f-error-message';
+import { FORM_EVENTS } from '../constants';
+
+export default {
+    components: {
+        ErrorMessage,
+        FormField
+    },
+
+    props: {
+        fieldData: {
+            type: Object,
+            required: true
+        }
+    },
+
+    inject: ['$v'],
+
+    computed: {
+        kebabCase () {
+            return this.fieldData.name.replace(/([a-z0-9]|(?=[A-Z]))([A-Z1-9])/g, '$1-$2').toLowerCase();
+        },
+
+        translations () {
+            return {
+                ...this.fieldData.translations,
+                name: `${this.kebabCase}`,
+                errorName: `${this.kebabCase}-error`,
+                emptyError: `error-${this.kebabCase}-empty`,
+                invalidError: `error-${this.kebabCase}-invalid`
+            };
+        },
+
+        fieldProps () {
+            let props = {
+                name: this.translations.name,
+                value: this.fieldData.value,
+                'label-text': this.translations.label,
+                'input-type': this.inputType,
+                'max-length': this.fieldData.mexLength || null
+            };
+
+            if (this.ariaLabel) {
+                props = {
+                    ...props,
+                    'aria-label': this.ariaLabel
+                };
+            }
+
+            if (this.hasError) {
+                props = {
+                    ...props,
+                    'has-error': true,
+                    'aria-invalid': true,
+                    'aria-describedby': this.translations.validationMessages[this.errorType]
+                };
+            }
+
+            return props;
+        },
+
+        inputType () {
+            const inputTypes = {
+                mobileNumber: 'tel',
+                email: 'email',
+                default: 'text'
+            };
+
+            return inputTypes[this.fieldData.name] || inputTypes.default;
+        },
+
+        hasValidationMessages () {
+            return !!this.translations?.validationMessages;
+        },
+
+        hasInvalidErrorMessage () {
+            return !!this.translations?.validationMessages?.invalid;
+        },
+
+        hasError () {
+            return !!this.hasValidationMessages && (this.hasRequiredError || this.hasInvalidError);
+        },
+
+        isEmpty () {
+            return this.validations?.$dirty;
+        },
```

Comment:
I'm not sure this makes sense to me, a field can be dirty but not empty

Reply:
Below it seems to be being used as if it were `isDirty`, so maybe just a rename will suffice?

Reply:
That makes a lot more sense, thanks